src/Lykke.Service.HFT.Wamp/Modules/WampModule.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Autofac;
+using Autofac.Core;
+using Lykke.Service.HFT.Core;
+using Lykke.Service.HFT.Core.Services.ApiKey;
+using Lykke.Service.HFT.Wamp.Auth;
+using Lykke.Service.HFT.Wamp.Consumers;
+using WampSharp.V2;
+using WampSharp.V2.Authentication;
+
+namespace Lykke.Service.HFT.Wamp.Modules
+{
+    public class WampModule : Module
+    {
+        private readonly AppSettings.HighFrequencyTradingSettings _settings;
+
+        public WampModule(AppSettings.HighFrequencyTradingSettings settings)
+        {
+            _settings = settings;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.RegisterType<WampSessionAuthenticatorFactory>()
+                .WithParameter(
+                    new ResolvedParameter(
+                        (pi, ctx) => pi.ParameterType == typeof(IApiKeyValidator),
+                        (pi, ctx) => ctx.Resolve<IApiKeyValidator>()))
+                .WithParameter(
```

Comment:
For what you define "WithParameter" in such registrations if they will be resolved anyway?

Reply:
That's a strange code indeed. I'll fix it.